packages/menu-bar/src/vaadin-menu-bar-mixin.js

```diff
@@ -276,6 +276,13 @@ export const MenuBarMixin = (superClass) =>
         }
         const items = buttons.filter((_, idx) => idx >= i).map((b) => b.item);
         this.__updateOverflow(items);
+
+        const remaining = buttons.slice(0, i);
+        // Ensure there is at least one button with tabindex set to 0
+        // so that menu-bar is not skipped when navigating with Tab
+        if (i > 0 && !remaining.some((btn) => btn.getAttribute('tabindex') === '0')) {
```

Comment:
nit: I am wondering if on the previous for loop, we could check for each button going to the overflow if they have the `tabindex` attribute set to `0` so that it can be used to determine if the `tabindex` needs to be set to the last visible button in case it was found. This way, we don't need to slice and use `some` to check the `tabindex`.

Reply:
I was thinking about that. However, before focusing menu-bar the first time, all buttons have `tabindex` set to `0`. So we can't rely on the attribute value on one button, as it might lead to wrong assumptions.